### ISSUE TITLE
Cannot find relative presets defined in `.babelrc`

### ISSUE THREAD

**Issue by vjpr** [storybooks/storybook#2771]
### Issue details

```
ERROR in ./.storybook/addons.js
Module build failed: Error: Couldn't find preset "./packages/babel-preset-foo" relative to directory "/Users/Vaughan/dev-live/foo/.storybook"
```

Project tree:

```
- .babelrc
- packages
  - babel-preset-foo
```

_.babelrc_

```
{"presets": ["./packages/babel-preset-foo"]}
```

### Please specify which version of Storybook and optionally any affected addons that you're running

```
    "@storybook/react": "^3.3.9",
    "@storybook/addon-actions": "^3.3.9",
    "@storybook/addon-links": "^3.3.9",
    "@storybook/addons": "^3.3.9"
```

**Comment by vjpr** [comment]
Simple fix was to add a `.babelrc` file to `.storybook` and correct the relative preset.

**Comment by vjpr** [comment]
This didn't actually solve my problem. It appears that babel tries to read relative presets from the dir of the file it is processing. So the preset path would need to be absolute, but this is not possible because `.babelrc` is static...

**Comment by vjpr** [comment]
Seems related to https://github.com/storybooks/storybook/issues/172

**Comment by vjpr** [comment]
The extends functionality should also be added for presets and plugins perhaps:

https://github.com/storybooks/storybook/blob/83d55c23fa124fa87cd061bdb5b5ce9453b393c1/app/vue/src/server/babel_config.js#L58-L66

**Comment by Hypnosphi** [comment]
Also related: #2610 #2582

**Comment by Hypnosphi** [comment]
Released as [`4.0.0-alpha.20`](https://github.com/storybooks/storybook/releases/tag/v4.0.0-alpha.20)
